### PR TITLE
Use TCS properly in AmqpTransportHandler and misc code clean-up.

### PR DESCRIPTION
**Pull request by drwill-ms**

### CODE REVIEW

File: iothub/device/src/Pipeline/DefaultDelegatingHandler.cs [hunk 1]
```diff
@@ -154,7 +154,7 @@ protected void ThrowIfDisposed()
             }
         }
 
-        protected virtual void Dispose(bool disposing)
+        protected private virtual void Dispose(bool disposing)
```
Comment (drwill-ms) [inline iothub/device/src/Pipeline/DefaultDelegatingHandler.cs:157]: We can use `protected private` in more places, for internal classes that have protected members.

File: iothub/device/src/Transport/Amqp/AmqpTransportHandler.cs [hunk 2]
```diff
@@ -448,20 +431,30 @@ private void TwinMessageListener(Twin twin, string correlationId, TwinCollection
...
+                            Logging.Info("Could not remove correlation Id to complete the task awaiter for a twin operation.", nameof(TwinMessageListener));
                     }
                 }
             }
         }
 
-        protected override void Dispose(bool disposing)
+        private void RemoveOldOperations(object _)
+        {
+            _ = _twinResponseTimeouts
```
Comment (drwill-ms) [inline iothub/device/src/Transport/Amqp/AmqpTransportHandler.cs:442]: Linq-ified.

File: iothub/device/src/Transport/Amqp/AmqpTransportHandler.cs [hunk 3]
```diff
@@ -392,9 +375,9 @@ public override async Task<long> SendTwinPatchAsync(TwinCollection reportedPrope
             try
             {
                 cancellationToken.ThrowIfCancellationRequested();
-                var taskCompletionSource = new TaskCompletionSource<Twin>();
+                var taskCompletionSource = new TaskCompletionSource<Twin>(TaskCreationOptions.RunContinuationsAsynchronously);
```
Comment (drwill-ms) [inline iothub/device/src/Transport/Amqp/AmqpTransportHandler.cs:378]: They key change I'm testing.

File: iothub/device/src/Transport/Amqp/AmqpTransportHandler.cs [hunk 4]
```diff
@@ -118,23 +118,6 @@ public override async Task OpenAsync(CancellationToken cancellationToken)
             }
         }
 
-        private void CheckTimeout(object _)
```
Comment (drwill-ms) [inline iothub/device/src/Transport/Amqp/AmqpTransportHandler.cs:121]: moved to "private" section,

File: iothub/device/src/Transport/Amqp/AmqpTransportHandler.cs (hunk 2 above)
Comment (timtay-microsoft) [reply iothub/device/src/Transport/Amqp/AmqpTransportHandler.cs:442]: We could make this same change in the MQTT layer as well @tmahmood-microsoft

**Review by timtay-microsoft — APPROVED** [review]

**Review by abhipsaMisra — APPROVED** [review]

File: iothub/device/src/Transport/Amqp/AmqpTransportHandler.cs (hunk 2 above)
Comment (drwill-ms) [reply iothub/device/src/Transport/Amqp/AmqpTransportHandler.cs:442]: I will when I get to that file.

**Pull request merged**